### PR TITLE
docs: lightclient dev guide setup and client creation

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

- Adding Setup page including:
    - Configuring `AppModuleBasic` for light client modules
    - How to create clients using core IBC and 02-client

closes: #XXXX

### Commit Message / Changelog Entry

```bash
type: docs: adding lightclient dev guide setup and client creation page
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/CONTRIBUTING.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/10-structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/CONTRIBUTING.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: docs/ibc/light-clients/setup.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+// GetQueryCmd performs a no-op. Please see the 02-client cli commands.
+func (AppModuleBasic) GetQueryCmd() *cobra.Command {
+	return nil
+}
+```
+
+## Creating clients
+
+A client is created by executing a new `MsgCreateClient` transaction composed with a valid `ClientState` and initial `ConsensusState` encoded as protobuf `Any`s.
```
Comment (damiannolan) [inline docs/ibc/light-clients/setup.md:66]: Maybe we can add here that this step is normally done by a relayer although not strictly required.

File: docs/ibc/light-clients/setup.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+order: 4
+-->
+
+# Setup
+
+Learn how to confgure light client modules and create clients using core IBC and the `02-client` submodule. {synopsis}
+
+## Configuring a light client module
+
+An IBC light client module must implement the [`AppModuleBasic`](https://github.com/cosmos/cosmos-sdk/blob/main/types/module/module.go#L50) interface in order to register its concrete types against the core IBC interfaces defined in `modules/core/exported`. This is accomplished via the `RegisterInterfaces` method which provides the light client module with the opportunity to register codec types using the chain's `InterfaceRegistery`. Please refer to the [`07-tendermint` codec registration](https://github.com/cosmos/ibc-go/blob/7ae97694af558db2924e66ce4ce6527c03da89a5/modules/light-clients/07-tendermint/codec.go#L11).
```
Comment (chatton) [inline docs/ibc/light-clients/setup.md:11]: should we add a sentence saying that all other interface functions can safely perform a no-op if there is no additional logic required?

File: docs/ibc/light-clients/setup.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+The `AppModuleBasic` interface may also be leveraged to install custom CLI handlers for light client module users.
+
+<!-- TODO: link to how to configure in app.go -->
```
Comment (chatton) [inline docs/ibc/light-clients/setup.md:15]: currently the doc should be able to be accessed via `../itegration.md#integrating-light-clients`. once #2905 is merged.

Maybe we can link there, and add a section here that simply says to register the light client module alongside or in place of `ibctm.AppModuleBasic{}`

File: docs/ibc/light-clients/setup.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+  google.protobuf.Any client_state = 1 [(gogoproto.moretags) = "yaml:\"client_state\""];
+  // consensus state associated with the client that corresponds to a given
+  // height.
+  google.protobuf.Any consensus_state = 2 [(gogoproto.moretags) = "yaml:\"consensus_state\""];
+  // signer address
+  string signer = 3;
+}
+```
+
+Leveraging protobuf `Any` encoding allows core IBC to [unpack](https://github.com/cosmos/ibc-go/blob/7ae97694af/modules/core/keeper/msg_server.go#L28-L36) both the `ClientState` and `ConsensusState` into their respective interface types registered previously using the light client module's `RegisterInterfaces` method.
```
Comment (chatton) [inline docs/ibc/light-clients/setup.md:86]: should we link to a tag instead of commit?

File: docs/ibc/light-clients/setup.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+Within the `02-client` submodule, the [`ClientState` is then initialized](https://github.com/cosmos/ibc-go/blob/7ae97694af/modules/core/02-client/keeper/client.go#L30-L34) with its own isolated key-value store, namespaced using a unique client identifier.
```
Comment (chatton) [inline docs/ibc/light-clients/setup.md:88]: same commit -> tag

File: docs/ibc/light-clients/setup.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+ 
+In order to successfully create an IBC client using a new client type it [must be supported](https://github.com/cosmos/ibc-go/blob/7ae97694af/modules/core/02-client/keeper/client.go#L18-L24). Light client support in IBC is gated by on-chain governance. The allow list may be updated by submitting a new governance proposal to update the `02-client` parameter `AllowedClients`.
+
+<!-- TODO: update when params are managed by ibc-go -->
```
Comment (chatton) [inline docs/ibc/light-clients/setup.md:92]: should we create an issue for this (or link to an existing one)

**Review by chatton — COMMENTED** [review]
Looks great! left a few small comments

File: docs/ibc/light-clients/setup.md [hunk 7]
```diff
@@ -0,0 +1,110 @@
+<!-- 
+order: 4
+-->
+
+# Setup
+
+Learn how to confgure light client modules and create clients using core IBC and the `02-client` submodule. {synopsis}
```
Comment (charleenfei) [inline docs/ibc/light-clients/setup.md:7]:
```suggestion
Learn how to configure light client modules and create clients using core IBC and the `02-client` submodule. {synopsis}
```

File: docs/ibc/light-clients/setup.md (hunk 1 above)
Comment (charleenfei) [reply docs/ibc/light-clients/setup.md:66]: i think that's a good idea -- kind of related to the comment I made in the planning meeting

File: docs/ibc/light-clients/setup.md (hunk 2 above)
Comment (damiannolan) [reply docs/ibc/light-clients/setup.md:11]: done!

File: docs/ibc/light-clients/setup.md (hunk 3 above)
Comment (damiannolan) [reply docs/ibc/light-clients/setup.md]: updated! :)

File: docs/ibc/light-clients/setup.md (hunk 4 above)
Comment (damiannolan) [reply docs/ibc/light-clients/setup.md]: I was thinking we will probably need to address all these links to point to a v7 final release when its tagged. I've updated to use the `02-client-refactor-beta1` tag for now

File: docs/ibc/light-clients/setup.md (hunk 6 above)
Comment (damiannolan) [reply docs/ibc/light-clients/setup.md]: Good call, added an issue link

File: docs/ibc/light-clients/setup.md (hunk 1 above)
Comment (damiannolan) [reply docs/ibc/light-clients/setup.md:67]: Yup! That sparked the thought 💡

File: docs/ibc/light-clients/client-state.md [hunk 8]
```diff
@@ -18,7 +18,7 @@ The format is created as follows: `ClientType-{N}` where `{N}` is the unique glo
 ## `Validate` method
 
 `Validate` should validate every client state field and should return an error if any value is invalid. The light client
-implementor is in charge of determining which checks are required. See the [tendermint light client implementation](https://github.com/cosmos/ibc-go/blob/v6.0.0-rc1/modules/light-clients/07-tendermint/types/client_state.go#L101)
+implementor is in charge of determining which checks are required. See the [tendermint light client implementation](https://github.com/cosmos/ibc-go/blob/v6.0.0/modules/light-clients/07-tendermint/types/client_state.go#L101)
```
Comment (charleenfei) [inline docs/ibc/light-clients/client-state.md:21]:
```suggestion
implementer is in charge of determining which checks are required. See the [tendermint light client implementation](https://github.com/cosmos/ibc-go/blob/v6.0.0/modules/light-clients/07-tendermint/types/client_state.go#L101)

**Review by charleenfei — APPROVED** [review]
just two more small nits but otherwise lgtm!

File: docs/ibc/light-clients/setup.md [hunk 9]
```diff
@@ -0,0 +1,121 @@
+<!-- 
+order: 4
+-->
+
+# Setup
+
+Learn how to configure light client modules and create clients using core IBC and the `02-client` submodule. {synopsis}
```
[the remark this answers is not in the document]
Comment (damiannolan) [reply docs/ibc/light-clients/setup.md:7]: Unsure if your question is about what `{synopsis}` does or the contents? 

If you render the docs with `make view-docs` the `{synopsis}` directive is what makes the little grey box with "Synopsis" in it, it's used in a lot of our docs pages.

[the remark this answers is not in the document]
Comment (charleenfei) [reply docs/ibc/light-clients/setup.md:7]: Oh oops! I didn't realise 🙈 this makes sense

**Pull request merged**